src/validation_functions.jl

```diff
@@ -0,0 +1,132 @@
+mutable struct validation_info
+    field_descriptor::Union{Nothing, Dict}
+    struct_name::AbstractString
+    validation_action::Union{Nothing, Bool}
+    component::PowerSystemType
+    field_type::Any
+    valid_min::Union{Nothing, Float64}
+    valid_max::Union{Nothing, Float64}
+end
+
+#Get validation info for one struct
+function get_config_descriptor(config::Vector, name::AbstractString)
+    for item in config
+        if item["struct_name"] == name
+            return item
+        end
+    end
+    error("component $name does not exist in validation configuration file")
+end
+
+#Get validation info for one field of one struct
+function get_field_descriptor(struct_descriptor::Dict, fieldname::AbstractString)
+    for field in struct_descriptor["fields"]
+        if field["name"] == fieldname
+            return field
+        end
+    end
+    error("field $fieldname does not exist in $(struct_descriptor["name"])")
+end
+
+function validate_fields(sys::System, component::T) where T <: PowerSystemType
+    struct_descriptor = get_config_descriptor(sys.validation_descriptor, repr(T))
+    valid_info = validation_info(nothing, struct_descriptor["struct_name"], false,
+                                           component, nothing, nothing, nothing)
+    for (name,fieldtype) in zip(fieldnames(T), fieldtypes(T))
+        field_value = getfield(component, name)
+        valid_info.field_type = fieldtype
+
+        if field_value == nothing
+            continue
+        end
+        if fieldtype <: Union{Nothing, PowerSystemType} && !(fieldtype <: Component)
+            error_detected = validate_fields(sys, getfield(component, name))
+            if error_detected
+                valid_info.validation_action = true
+            end
+        else
+            field_descriptor = get_field_descriptor(struct_descriptor, string(name))
+            valid_info.field_descriptor = field_descriptor
+            validate_num_value(valid_info, field_value)
+        end
+    end
+    return valid_info.validation_action
+end
+
+function validate_num_value(valid_info::validation_info, field_value)
+    if haskey(valid_info.field_descriptor, "valid_range")
+        if valid_info.field_descriptor["valid_range"] isa String
```

Comment:
Did you consider dispatching on the type of valid_info.field_descriptor["valid_range"]?  You could define the function validate_range with two methods.  One could accept a dict (maybe even a {::Min_Max}, one could accept string, others could add methods in the future.